sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+        deflection_curve = integrate((E*I)*self.slope(), x) + C4
+
+        bc_eqs = []
+        for position, value in self._boundary_conditions['deflection']:
+            eqs = deflection_curve.subs(x, position) - value
+            bc_eqs.append(eqs)
+
+        constants = list(linsolve(bc_eqs, C4))
+        deflection_curve = deflection_curve.subs({C4 : constants[0][0]})
+        return S(1)/(E*I)*deflection_curve
+
+
+class PointLoad(object):
+    """A load (force or moment) applied at a point.
+
+    A load applied to a single, specific point. It is also known as a
+    concentrated load.
+
+    Parameters
+    ==========
+    location : Sympifyable
+        The specific point of the applied load.
+    value : Sympifyable
+        A SymPy expression representing the value of the applied load.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import PointLoad
+    >>> PointLoad(location = 4, value = -4)
+    PointLoad(4, -4, Load)
+
+    A Moment can be defined just by passing moment=True as an argument.
+
+    >>> PointLoad(location = 4, value = -4, moment=True)
+    PointLoad(4, -4, Moment)
+
+    """
+
+    def __init__(self, location, value, moment=False):
+        self._location = location
+        self._value = value
+        self._moment = moment
+
+    def __str__(self):
+        if self.moment:
+            str_sol = 'PointLoad(%s, %s, %s)' % (sstr(self.location), sstr(self.value), sstr('Moment'))
+            return str_sol
+        str_sol = 'PointLoad(%s, %s, %s)' % (sstr(self.location), sstr(self.value), sstr('Load'))
+        return str_sol
+
+    __repr__ = __str__
+
+    @property
+    def location(self):
+        """Location of the applied Point Load."""
+        return self._location
+
+    @location.setter
+    def location(self, l):
+        self._location = sympify(l)
+
+    @property
+    def value(self):
+        """Value of the applied Point Load. """
+        return self._value
+
+    @value.setter
+    def value(self, v):
+        self._value = sympify(v)
+
+    @property
+    def moment(self):
+        """Stores whether a Point Load is a load or a couple. """
+        return self._moment
+
+    @moment.setter
+    def moment(self, m):
+        if not isinstance(m, bool):
+            raise TypeError("PointLoad moment attribute must be a bool object.")
+        self._moment = m
+
+
+class DistributedLoad(object):
+
+    """A Distributed Load.
+
+    A load applied across a length instead of at one point.
+
+    Parameters
+    ==========
+    value : Sympifyable
+        A SymPy expression representing the value of the applied load.
+    start : Sympifyable
+        The starting point of the applied load.
+    order : Sympifyable
+        The order of the applied load.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import DistributedLoad
+    >>> DistributedLoad(start = 4, order = 2, value = 2)
+    DistributedLoad(2, 4, 2)
+
+    """
```

Comment:
Attributes and methods can be described in the class docstring, as well as general descriptions of the class. Input args/kwargs to the classes methods should be describe in the method docstring. This applies to `__init__` and any other public, private, or builtin methods. The `__init__` method docstring should show how to initialize the class.